biothings/hub/databuild/backend.py

```diff
@@ -157,23 +157,27 @@ def get_src_metadata(self):
 
             if src and src.get("download"):
                 # Store the latest success dump time
-                if src["download"]["status"] == "success":
-                    src_meta.setdefault(src["_id"], {}).setdefault("download_date", src["download"]["started_at"])
+                src_meta.setdefault(src["_id"], {})
+                src_meta[src["_id"]].setdefault(
+                    "download_date", src["download"].get("last_success")
+                )
 
             if src and src.get("upload"):
                 latest_upload_date = None
                 meta = {}
                 for job_name in src["upload"].get("jobs", {}):
                     job = src["upload"]["jobs"][job_name]
                     # "step" is the actual sub-source name
-                    docm = self.master.find_one({"_id": job.get("step")})
+                    sub_source = job.get("step")
+                    docm = self.master.find_one({"_id": sub_source})
                     if docm and docm.get("src_meta"):
-                        meta[job.get("step")] = docm["src_meta"]
+                        meta[sub_source] = docm["src_meta"]
                     # Store the latest success upload time
                     if not latest_upload_date or latest_upload_date < job["started_at"]:
-                        if job["status"] == "success":
-                            latest_upload_date = job["started_at"]
-                            meta[job.get("step")]["upload_date"] = latest_upload_date
+                        step_meta = meta.setdefault(sub_source, {})
+                        step_meta["upload_date"] = src["upload"]["jobs"][sub_source].get(
+                            "last_success"
+                        )
```

Comment:
@sengineer0 same case as the above comment.

Reply:
ok sure